### ISSUE TITLE
tty-clock is late

### ISSUE THREAD

**Issue by anarcat** [xorg62/tty-clock#40]
When compared with other software, tty-clock seems actually to run one or two seconds late.

![snap-2017 01 13-10 27 23](https://cloud.githubusercontent.com/assets/796623/21935285/f557a7d8-d97a-11e6-854a-9a4fab552ee0.png)

In the above screenshot, 4 clocks are running, from top to bottom:

1. xmobar (the status bar of xmonad)
2. `xclock -d -update 1`
3. [`vtclock -v -3`](https://github.com/dse/vtclock/) (nothing special about those flags, just a nicer font)
4. `tty-clock -s`

Note that we don't see it in the screenshot, but xmobar is actually *lagging* a little behind the other two clocks, say about 500ms. Here's a [screencast](https://github.com/xorg62/tty-clock/files/704652/out.ogv.gz) that shows the behavior more precisely.

I find it mind-boggling that 4 clocks can't agree on the time on my computer, this gotta have some profound meaning involving long hair and swiss patent offices, but i am not certain... ;)

**Comment by anarcat** [comment]
for the record:

1. xmobar and taffybar just refresh the display whenever it feels like it and checks the actual time to display it
2. xclock does some [funky arithmetic](http://sources.debian.net/src/x11-apps/7.7%2B6/xclock/Clock.c/#L1251) to figure when to wakeup
3. vtclock does a [busyloop on gettimeofday](https://github.com/dse/vtclock/blob/master/vtclock.c#L125)
4. ttyclock just sleeps the configured time then calls the localtime on the *previous* time set, then sets the time (!!)

```
     ttyclock->tm = localtime(&(ttyclock->lt));
     if(ttyclock->option.utc) {
         ttyclock->tm = gmtime(&(ttyclock->lt));
     }
     ttyclock->lt = time(NULL);
```

i'll send a pr to fix this. :) with the patch, the display is more consistent with xclock and friends.

**Comment by anarcat** [comment]
should be fixed in #41
